clusterloader2/pkg/measurement/common/waitforpods.go

```diff
@@ -0,0 +1,99 @@
+/*
+Copyright 2018 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package common
+
+import (
+	"fmt"
+	"time"
+
+	"github.com/golang/glog"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/perf-tests/clusterloader2/pkg/measurement"
+	"k8s.io/perf-tests/clusterloader2/pkg/util"
+)
+
+func init() {
+	measurement.Register("WaitForPods", createWaitForPodMeasurment)
+}
+
+func createWaitForPodMeasurment() measurement.Measurement {
+	return &waitForPodMeasurment{}
+}
+
+type waitForPodMeasurment struct{}
```

Comment:
waitForPod what? For e.g it could be waitForPodRunning, waitForPodReady, etc. I suggest taking that as a parameter in the function and  deciding logic based on it.

Reply:
Also I suggest waitForPod's' - as we're waiting for multiple pods.

Reply:
Change to `waitForRunningPodsMeasurement`.
It don't want to allow (at least not now) passing pod status as parameter - handling for this doesn't seem trivial.